src/trace.rs

```diff
@@ -271,13 +271,76 @@ impl<S> BroadcastLayer<S> {
 
 impl<S> Layer<S> for BroadcastLayer<S>
 where
-    S: Subscriber + 'static,
+    S: Subscriber + 'static + for<'lookup> LookupSpan<'lookup>,
 {
-    fn on_event(&self, event: &Event<'_>, _ctx: Context<'_, S>) {
-        // Try buffering the event, and if we're not buffering anymore, try to send it along via the
-        // trace sender if it's been established.
-        if !try_buffer_event(event) {
-            try_broadcast_event(event);
+    fn on_event(&self, event: &Event<'_>, ctx: Context<'_, S>) {
+        let mut log = LogEvent::from(event);
+        // Add span fields if available
+        if let Some(parent_span) = ctx.event_span(event) {
+            for span in parent_span.scope().from_root() {
+                if let Some(fields) = span.extensions().get::<SpanFields>() {
+                    for (k, v) in &fields.0 {
+                        log.insert(format!("vector.{}", k).as_str(), v.clone());
```

Comment:
Note, these fields are now placed in a `vector` namespace given the conversation that motivated this separate ticket: https://datadoghq.atlassian.net/browse/OP-398